pkg/iam/policy/policy.go

```diff
@@ -48,11 +48,13 @@ type Policy struct {
 // IsAllowed - checks given policy args is allowed to continue the Rest API.
 func (iamp Policy) IsAllowed(args Args) bool {
 	// Check all deny statements. If any one statement denies, return false.
+	var deny = true // Deny is true by default, unless request is invalidated.
```

Comment:
From the AWS docs on [policy-evaluation-logic](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_evaluation-logic.html), if we set `deny` to false in the absence of an explicit deny and allow a matching request, then we would be violating the **implicit deny principle**.

Reply:
No @krisis it's only after validating Deny statement we never even go into a Deny statement if the Deny statement doesn't exist. If there are no deny statements and allowed is also false we don't allow the request. are you saying we allow after this change?

Reply:
Thanks looks like you are right, changed the code accordingly. We don't have a bug in our implementation.